.circleci/config.yml

```
@@ -0,0 +1,181 @@
+version: 2
+
+workflows:
+  version: 2
+  lint_build_test:
+    jobs:
+      - lint
+      - emu_gcc_ninja:
+          requires:
+            - lint
+      - emu_gcc_ninja_debug:
+          requires:
+            - lint
+      - emu_gcc_make:
+          requires:
+            - lint
+      - emu_clang_ninja:
+          requires:
+            - lint
+      - kindle:
+          requires:
+            - emu_gcc_ninja_debug
+      - kobo:
+          requires:
+            - emu_gcc_ninja_debug
+      - pocketbook:
+          requires:
+            - emu_gcc_ninja_debug
+      - sony-prstux:
+          requires:
+            - emu_gcc_ninja_debug
+      - cervantes:
+          requires:
+            - emu_gcc_ninja_debug
+      - android-arm:
+          requires:
+            - emu_gcc_ninja_debug
+      - android-x86:
+          requires:
+            - emu_gcc_ninja_debug
+
+
+jobs:
+  lint:
+    docker:
+      - image: frenzie/kobase:0.1.2
+    environment:
+      BASH_ENV: "~/.bashrc"
+    steps:
+      - checkout
+      - run:
+          name: install
+          command: |
+            source .ci/lint_install.sh
+      - run:
+          name: lint
+          command: |
+            source .ci/lint_script.sh
+
+  emu: &EMU_TPL
+    docker:
+      - image: frenzie/kobase:0.1.2
+    environment: &EMU_ENV_LST
+      BASH_ENV: "~/.bashrc"
+      EMULATE_READER: "1"
+      CC: "gcc"
+    steps:
+      - checkout
+      - run:
+          name: install
+          command: |
+            source .ci/before_install.sh
+            source .ci/emulator_install.sh
+
+      - restore_cache:
+          keys:
+            - x86-{{ .Environment.CIRCLE_JOB }}-
```

```suggestion
            - {{ arch }}-{{ .Environment.CIRCLE_JOB }}-
```
See https://github.com/koreader/koreader/pull/3329

---

i actually didn't intend to put the actual arch here, just a prefix to distinguish between `xcompile-` jobs. since we are only using linux runners here, `arch` should be the same for all jobs right?

i changed the prefix to `emu` so it's less confusing :)